homeassistant/components/sensor/whois.py

```diff
@@ -99,38 +95,51 @@ def state(self):
     @property
     def device_state_attributes(self):
         """Get the more info attributes."""
-        if self._data:
-            updated_formatted = self._updated_date.isoformat()
-            expires_formatted = self._expiration_date.isoformat()
+        return self._attributes
 
-            return {
-                ATTR_NAME_SERVERS: ' '.join(self._name_servers),
-                ATTR_REGISTRAR: self._data['registrar'][0],
-                ATTR_UPDATED: updated_formatted,
-                ATTR_EXPIRES: expires_formatted,
-            }
+    def _empty_state_and_error(self, error_message, *args):
+        """Empty the state and attributes on an error."""
+        self._state = None
+        self._attributes = None
+
+        _LOGGER.error(error_message, *args)
 
     def update(self):
-        """Get the current WHOIS data for hostname."""
+        """Get the current WHOIS data for the domain."""
         from pythonwhois.shared import WhoisException
 
         try:
             response = self.whois(self._domain, normalized=True)
         except WhoisException as ex:
-            _LOGGER.error("Exception %s occurred during WHOIS lookup", ex)
+            return self._empty_state_and_error(
+                "Exception %s occurred during WHOIS lookup", ex)
             return
```

Comment:
FYI, linter is complaining because of this extra `return` here.

Reply:
Aha, silly me 😀 Been down with a bug but will resolve this soon.